src/App.js

```diff
@@ -1,10 +1,33 @@
 import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
-import { Home, Login, Signup, Bookmark, AllPost, Profile } from "./pages";
+import {
+  Wrapper,
+  Login,
+  Signup,
+  Bookmark,
+  AllPost,
+  Profile,
+  Feed,
+} from "./pages";
 import { ToastContainer } from "react-toastify";
 import { RequireAuth } from "./RequiresAuth";
 import "react-toastify/dist/ReactToastify.css";
+import { PostCardM } from "./components";
+import { useDispatch, useSelector } from "react-redux";
+import { useEffect } from "react";
+import { getAllPost, getUserPost } from "./pages/home/postSlice";
+import { getUsers } from "./pages/profile/userSlice";
 
 const App = () => {
+  const dispatch = useDispatch();
+  const { token, user } = useSelector((state) => state.auth);
+
+  useEffect(() => {
+    if (token) {
+      dispatch(getAllPost());
+      dispatch(getUserPost(user.username));
+      dispatch(getUsers());
+    }
+  }, [token]);
   return (
     <div className="App">
       <ToastContainer

```

Comment:
You can move toast to a new component and use that component here instead of wiring up it here.